MatrixSDK/Background/MXBackgroundSyncService.swift

```diff
@@ -504,6 +504,21 @@ public enum MXBackgroundSyncServiceError: Error {
             handleToDeviceEvent(event)
         }
         
+        if MXSDKOptions.sharedInstance().autoAcceptRoomInvites,
+           let invitedRooms = syncResponse.rooms?.invite {
+            invitedRooms.forEach { roomId, roomSync in
+                MXLog.debug("[MXBackgroundSyncService] handleSyncResponse: Auto-accepting room invite for \(roomId)")
+                restClient.joinRoom(roomId) { response in
+                    switch response {
+                    case .success:
+                        MXLog.debug("[MXBackgroundSyncService] handleSyncResponse: Joined room: \(roomId)")
+                    case .failure(let error):
+                        MXLog.error("[MXBackgroundSyncService] handleSyncResponse: Failed to join room: \(roomId), error: \(error)")
```

Comment:
Do you know if the `/join` request generates an error if we double join a room?
This is what will happen if the auto-join is firstly managed by the NSE.

Reply:
You're right. I've checked this and seems we don't have an error. I need to fix this.

Reply:
Error added with https://github.com/matrix-org/matrix-ios-sdk/pull/1134/commits/260d68416d5d0c92914b48af08c2182468a8f898

Reply:
Retry mechanism added with https://github.com/matrix-org/matrix-ios-sdk/pull/1134/commits/93aca3b8481d8d606dee8bb99c2984e8a4182feb